nextcloud/server
# Add database ratelimiting backend

In case no distributed memory cache is specified this adds a database backend for ratelimiting purposes. Previously the ratelimit check was just skipped.

This also fixes an issue where we passed always the current time instead of the storage period to the backend. And simplified the API to not pass `$period` where not necessary.

This doesn't aim to be a high performance solution, but just a fallback for instances without any kind of memory cache configured.

# Test plan

- Disable the memory cache in `config.php`
- Enable the "Q&A Testing App" in the app settings
- Go to `/index.php/apps/testing/userAndAnonProtected` and click a few times. Verify that the requests fails due to hitting the ratelimit.
- Verify that expired ratelimit entries are removed on requests
- Repeat the same with memory cache enabled

# TODO

- [x] Revert unnecessary composer changes
- [x] Cleanup unneeded entries on check
- [x] PHP CS fixup
